**Importing Necessary Libraries and Data set**

In [9]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_diabetes
X, y = load_diabetes(return_X_y = True, as_frame = True)

In [25]:
##Checking 
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 442 entries, 0 to 441
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   age     442 non-null    float64
 1   sex     442 non-null    float64
 2   bmi     442 non-null    float64
 3   bp      442 non-null    float64
 4   s1      442 non-null    float64
 5   s2      442 non-null    float64
 6   s3      442 non-null    float64
 7   s4      442 non-null    float64
 8   s5      442 non-null    float64
 9   s6      442 non-null    float64
dtypes: float64(10)
memory usage: 34.7 KB


In [11]:
y

0      151.0
1       75.0
2      141.0
3      206.0
4      135.0
       ...  
437    178.0
438    104.0
439    132.0
440    220.0
441     57.0
Name: target, Length: 442, dtype: float64

In [12]:
y.shape

(442,)

In [13]:
##Merging x and y
diabetes = X.copy()
diabetes["disease_progress"] = y
diabetes

age       sex       bmi        bp        s1        s2        s3  \
0    0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401   
1   -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412   
2    0.085299  0.050680  0.044451 -0.005671 -0.045599 -0.034194 -0.032356   
3   -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038   
4    0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142   
..        ...       ...       ...       ...       ...       ...       ...   
437  0.041708  0.050680  0.019662  0.059744 -0.005697 -0.002566 -0.028674   
438 -0.005515  0.050680 -0.015906 -0.067642  0.049341  0.079165 -0.028674   
439  0.041708  0.050680 -0.015906  0.017282 -0.037344 -0.013840 -0.024993   
440 -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674   
441 -0.045472 -0.044642 -0.073030 -0.081414  0.083740  0.027809  0.173816   

           s4        s5        s6  disease_progress  
0   -0.002592  0.019908 -0.017646             151.0  
1   -0.039493 -0.068330 -0.092204              75.0  
2   -0.002592  0.002864 -0.025930             141.0  
3    0.034309  0.022692 -0.009362             206.0  
4   -0.002592 -0.031991 -0.046641             135.0  
..        ...       ...       ...               ...  
437 -0.002592  0.031193  0.007207             178.0  
438  0.034309 -0.018118  0.044485             104.0  
439 -0.011080 -0.046879  0.015491             132.0  
440  0.026560  0.044528 -0.025930             220.0  
441 -0.039493 -0.004220  0.003064              57.0  

[442 rows x 11 columns]

In [14]:
diabetes["disease_progress"].nunique()

214

**Building the Random Forest Model**

In [15]:
x=diabetes.drop("disease_progress", axis = 1)
y=diabetes["disease_progress"]

In [16]:
##aplitting data into test and train
from sklearn.model_selection import train_test_split
x_train, x_test, y_train,  y_test = train_test_split(x,y, test_size=0.3, random_state = 0)

In [17]:
##building the model
from sklearn.ensemble import RandomForestClassifier
model = RandomForestClassifier (n_estimators = 20)
model.fit(x_train,y_train)

RandomForestClassifier(n_estimators=20)

**Evaluating the model**

In [18]:
model.score(x_test, y_test)

0.0

In [19]:
y_pred = model.predict(x_test)

In [20]:
from sklearn.metrics import confusion_matrix
con_max = confusion_matrix(y_test, y_pred)
con_max

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

**Conclusion:** From the model evaluation, we can conclude that Random Forest Classifier is not suitable for this data set. We can see that the number of unique values in the target column(disease progress) is 214, and are numerical, hence, a linear regression model would perform better with the data 